vsphere/resource_vsphere_virtual_machine.go

```diff
@@ -283,6 +283,23 @@ func resourceVSphereVirtualMachine() *schema.Resource {
 			Computed:    true,
 			Description: "The power state of the virtual machine.",
 		},
+		"usb_controller": {
+			Type:        schema.TypeList,
+			Optional:    true,
+			Description: "A specification for a USB controller on the virtual machine.",
+			Elem: &schema.Resource{
+				Schema: map[string]*schema.Schema{
+					"usb_version": {
+						Type:         schema.TypeString,
+						Optional:     true,
+						Default:      "2.0",
+						Description:  "The version of the USB controller.",
+						ValidateFunc: validation.StringInSlice([]string{"2.0", "3.1"}, false),
+					},
+				},
+			},
+		},
```

Comment:
Maybe:

```go
			},
		},
		ValidateFunc: validateUSBControllerVersions,
```

And add a function to ensure only one of each can be added. FOr example

```go
func validateUSBControllerVersions(val interface{}, key string) (warns []string, errs []error) {
    usbControllers := val.([]interface{})
    versionCount := make(map[string]int)
    for _, version := range usbControllerVersions {
        versionCount[version] = 0
    }

    for _, controller := range usbControllers {
        controllerMap := controller.(map[string]interface{})
        version := controllerMap["usb_version"].(string)
        if _, exists := versionCount[version]; exists {
            versionCount[version]++
            if versionCount[version] > 1 {
                errs = append(errs, fmt.Errorf("%q cannot have more than one USB controller of version %s", key, version))
            }
        }
    }
    return
}
```